src/js/builtins/ProcessObjectInternals.ts

```diff
@@ -145,7 +147,11 @@ export function getStdinStream(fd) {
         }
       } else {
         stream.emit("end");
-        stream.pause();
+        if (!stream_destroyed) {
+          stream_destroyed = true;
+          stream.destroy();
+          unref();
+        }
```

Comment:
Why don't you emit 'close' to avoid to duplicate code and add boolean variable?

Reply:
because of the close handler below that.

Reply:
Why destroy() on a stream is emitting the 'close' event? 
Semantically, destroying is orthogonal to emitting an event. The destroy() caller is responsible of emitting event accordingly given the context.